File: fidouafclient/src/main/java/io/hanko/fidouafclient/client/msg/client/DiscoveryData.java
package io.hanko.fidouafclient.client.msg.client;

import io.hanko.fidouafclient.authenticator.msgs.Authenticator;
import io.hanko.fidouafclient.client.msg.Version;

public class DiscoveryData {
    public Version[] supportedUAFVersions;
    public String clientVendor;
    public Version clientVersion;
    public Authenticator[] availableAuthenticators;
}

File: fidouafclient/src/main/java/io/hanko/fidouafclient/client/msg/client/DiscoveryData.kt
package io.hanko.fidouafclient.client.msg.client

import com.squareup.moshi.JsonClass
import io.hanko.fidouafclient.authenticator.msgs.Authenticator
import io.hanko.fidouafclient.client.msg.Version

@JsonClass(generateAdapter = true)
class DiscoveryData (
    val supportedUAFVersions: List<Version>,
    val clientVendor: String,
    val clientVersion: Version,
    val availableAuthenticators: List<Authenticator>
)
